### PR TITLE
add functionality required to rotate certificates

### PR DESCRIPTION
**Pull request by deads2k**
1. filter expired certificates

@openshift/sig-auth fyi

WIP while I work through what is needed.

### CODE REVIEW

File: pkg/crypto/crypto.go [hunk 1]
```diff
@@ -454,6 +476,18 @@ func GetCA(certFile, keyFile, serialFile string) (*CA, error) {
 	}, nil
 }
 
+func GetCAFromBytes(certBytes, keyBytes []byte) (*CA, error) {
+	caConfig, err := GetTLSCertificateConfigFromBytes(certBytes, keyBytes)
```
Comment (ericavonb) [inline pkg/crypto/crypto.go]: It seems unnecessary to have both `GetCAFromBytes` and `GetTLSCertificateConfigFromBytes`. Can you simplify it before we merge?

**Comment by mrogers950** [comment]
@deads2k is this still a WIP? It looks OK to me at the moment.

**Pull request merged**